lib/Mojo/WebSocketProxy/Backend.pm

```diff
@@ -1,20 +1,45 @@
 package Mojo::WebSocketProxy::Backend;
-# TODO(leonerd): Split out JSON-RPC-specific parts into a subclass
 
 use strict;
 use warnings;
 
-use Mojo::Base -base;
+no indirect;
 
-use MojoX::JSON::RPC::Client;
-use Guard;
+use Mojo::Util qw(class_to_path);
 
 ## VERSION
 
-has 'url';
+our %CLASSES = (
```

Comment:
How would you feel about something like `Module::PluginFinder` here, rather than a statically configured list in one place?

Failing that, maybe have each of the specific modules register themselves into this base,

    package Mojo::WebSocketProxy::Backend::JSONRPC;
    use parent qw( Mojo::WebSocketProxy::Backend );
    __PACKAGE__->register_type( 'jsonrpc' );

That way, this common base doesn't need to know the names of specific types.

Reply:
The registration approach would be sensible, yes - I know some people are dubious about autoloading everything available and just want to restrict to a subset, but that `->register_type` option is much neater than the current version.

Reply:
Yes - `PluginFinder` will `require` every possible module found on the filesystem, whereas a toplevel main program that `use`s just the ones it wants would be fine.

Reply:
Done 24f7ec3